packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -137,24 +135,13 @@ contract TaikoL2 is CrossChainOwned, ICrossChainSync {
         // Verify the base fee per gas is correct
         uint256 basefee;
         (basefee, gasExcess) = _calc1559BaseFee(config, l1Height, parentGasUsed);
-        if (!skipFeeCheck() && block.basefee != basefee) {
-            revert L2_BASEFEE_MISMATCH();
-        }
 
-        // Store the L1's state root as a signal to the local signal service to
-        // allow for multi-hop bridging.
-        ISignalService(resolve("signal_service", false)).sendSignal(l1StateRoot);
+        verifyBaseFee(basefee);
 
-        emit CrossChainSynced(uint64(block.number), l1Height, l1BlockHash, l1StateRoot);
+        relayStateRoot(ownerChainId, l1StateRoot);
 
         // Update state variables
         l2Hashes[parentId] = blockhash(parentId);
-        snippets[l1Height] = ICrossChainSync.Snippet({
-            remoteBlockId: l1Height,
-            syncedInBlock: uint64(block.number),
-            blockHash: l1BlockHash,
-            stateRoot: l1StateRoot
-        });
```

Comment:
I also realized the differences. I think it's fine as most apps shall not assuming these data are available otherwise, the app will not be deployable on other chains. 

But I do think it may be benefit to add the latest cached data and its block number to TaikoL1/L2 contract, and the cost is very small (writing to the same slot).

Reply:
Indeed hard to predict how useful it will be.

But it all depends on what other L2s also make available, because the function to call to get the data will be different, but the dapp could depend on the data to be there easily by just pointing to the right contract for each chain. I haven't look too closely to what is available on optimism, but there at least the full block header of the last known L1 block is made available in a contract: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/L1Block.sol. That means that dapps that depend on this information can run on Optimism but would not be able to run on Taiko.